x/observer/keeper/msg_server_vote_tss.go

```diff
@@ -94,6 +89,21 @@ func (k msgServer) VoteTSS(goCtx context.Context, msg *types.MsgVoteTSS) (*types
 		}, nil
 	}
 
+	// The ballot is finalized, we check if this is the correct ballot for updating the TSS
+	// The requirements are
+	// 1. The keygen is still pending
+	// 2. The keygen block number matches the ballot block number ,which makes sure this the correct ballot for the current keygen
+
+	// Return without error so the vote is added to the ballot
+	if keygen.Status != types.KeygenStatus_PendingKeygen {
+		return &types.MsgVoteTSSResponse{}, nil
+	}
+
+	// For cases when an observer tries to vote for an older pending ballot , associated with a keygen that was discarded , we would return at this check while still adding the vote to the ballot
+	if msg.KeygenZetaHeight != keygen.BlockNumber {
```

Comment:
Should this be `msg.KeygenZetaHeight > keygen.BlockNumber`?

Reply:
The check is added so that we can add votes to an older ballot. 
When that happens the `msg.KeygenZetaHeight ` would always be less than the `keygen.BlockNumber` .

We could modify this to `msg.KeygenZetaHeight < keygen.BlockNumber` but I think using `!=` is fine as well